autocodesign/localcodesignasset/profilelookup.go

```diff
@@ -0,0 +1,121 @@
+	if !isActive(profile, minProfileDaysValid) {
+		return false
+	}
+
+	if !hasMatchingDistributionType(profile, distributionType) {
+		return false
+	}
+
+	if !hasMatchingBundleID(profile, bundleID) {
+		return false
+	}
+
+	if !hasMatchingPlatform(profile, platform) {
+		return false
+	}
+
+	if !hasMatchingLocalCertificate(profile, certSerials) {
+		return false
+	}
+
+	if !containsAllAppEntitlements(profile, entitlements) {
+		return false
+	}
+
+	if !provisionsDevices(profile, deviceIDs) {
+		return false
+	}
+
+	return true
+}
+
+func hasMatchingBundleID(profile profileutil.ProvisioningProfileInfoModel, bundleID string) bool {
+	return profile.BundleID == bundleID
+}
+
+func hasMatchingLocalCertificate(profile profileutil.ProvisioningProfileInfoModel, localCertificateSerials []string) bool {
+	var profileCertificateSerials []string
+	for _, certificate := range profile.DeveloperCertificates {
+		profileCertificateSerials = append(profileCertificateSerials, certificate.Serial)
+	}
+
+	return 0 < len(intersection(localCertificateSerials, profileCertificateSerials))
+}
+
+func containsAllAppEntitlements(profile profileutil.ProvisioningProfileInfoModel, appEntitlements autocodesign.Entitlements) bool {
+	profileEntitlements := autocodesign.Entitlements(profile.Entitlements)
+	hasMissingEntitlement := false
+
+	for key, value := range appEntitlements {
+		profileEntitlementValue := profileEntitlements[key]
+
+		// The project entitlement values can have variables coming from build settings which will be resolved later
+		// during the archive action. It is not the best but this is also the logic used at other places. An example of
+		// what we could be comparing:
+		// 		$(AppIdentifierPrefix)${BASE_BUNDLE_ID}.ios == 72SA8V3WYL.io.bitrise.samples.fruta.los
+		if reflect.DeepEqual(profileEntitlementValue, value) == false {
+			hasMissingEntitlement = true
+			break
+		}
+	}
+
+	return !hasMissingEntitlement
+}
+
+func hasMatchingDistributionType(profile profileutil.ProvisioningProfileInfoModel, distributionType autocodesign.DistributionType) bool {
+	return autocodesign.DistributionType(profile.ExportType) == distributionType
+}
+
+func isActive(profile profileutil.ProvisioningProfileInfoModel, minProfileDaysValid int) bool {
+	expiration := time.Now()
+	if minProfileDaysValid > 0 {
+		expiration = expiration.AddDate(0, 0, minProfileDaysValid)
+	}
+
+	return expiration.Before(profile.ExpirationDate)
+}
+
+func hasMatchingPlatform(profile profileutil.ProvisioningProfileInfoModel, platform autocodesign.Platform) bool {
+	return strings.ToLower(string(platform)) == string(profile.Type)
+}
+
+func provisionsDevices(profile profileutil.ProvisioningProfileInfoModel, deviceIDs []string) bool {
+	if profile.ProvisionsAllDevices || len(deviceIDs) == 0 {
```

Comment:
NOTE: Still correct, but checking ProvisionsAllDevices should not change the outcome as deviceIDs is present only for ad-hoc and development at this point. 
ProvisionsAllDevices is set for Enterprise and unset for app-store (https://github.com/bitrise-io/go-xcode/blob/13f817b37b1c63bfa16ae226f4389f026accfe87/profileutil/plist_data.go#L105)